Python/PR_Analyzer/README.md

```diff
@@ -0,0 +1,100 @@
+  * We need to find out all the distinct contributors where every PR can involve more than one contributor.
+  * For that we can write a nested for loop to append only if the string is different from all the strings which were added to the list in before iterations.To refer codebase, click [[here](https://github.com/Aditya-Komaravolu/PR-Analyzer/blob/main/PRScrapper.ipynb)].
+  * Once done, we now have to compute the total number of contributions were done by each contributor in all the PR's. 
+  * To calculate the count, define a for loop which tallies each and every contributor with the the actual column in the DataFrame.
+  
+ #### Step 6: Create a JSON File
+  * WE can store the contributors and the count as key-value pairs in a dictionary.
+  * Later use file pointer and json package to dump the dictionary as a json object.
+ 
+ #### Step 7: Visualizing Data
+  * Now that we have PR status and Total number of contributions by each contributor, we can plot them for better visualization of the entire repo.
+  * I prefer `Bar Plots` for the current situation because we have categorical variables to represent and what else is better than a bar graph!  :)
+
+ #### <u>Additional Analysis</u>: Finding out the Average time taken for each Pull Request
+ 
+ * To find out the start and end time, locate the time reference for every container in the DevTools.
+  
+   ![image](https://user-images.githubusercontent.com/64011471/131406319-db2f7d6a-9467-4bf2-8e5b-24773ed79c10.png)
+ 
+ * Once we get the tag associated with the datetime, we can reapeat step 2 to get complete dates along with timestamp from every PR.
+ 
+   ![image](https://user-images.githubusercontent.com/64011471/131406692-f3e690cf-0e3b-4971-9d22-9b1947806081.png)
+
+ * Take minimum and maximum values from all the timestamps from every PR and subtract them to get the time taken to merge a PR.
+ * If you want, you can convert it into Days H:M:S format by using the datetime package to convert the string obejct to a datetime object.
+ * Datetime objects can be converted into seconds format by using `total_seconds`.
+ 
+   ![image](https://user-images.githubusercontent.com/64011471/131408733-dc0eec0b-6843-41e1-8b79-50a2e42a69c4.png)
+
+ 
+ #### Categorizing the Active PR time using KMeans Clustering
+ * Now that we have the active PR time in seconds format, it is now easier to fit the data to model.
+ * I choose KMeans because here we have active PR time in the form of clusters (atleast as per my initial observation) and since we are about to deal with uncategorised Pr time values, what's better than KMeans.
+ * Pass the single-column 'Active PR Time(in Seconds)' to the model with `n_clusters=3`.
+ * Later plot the centriods of those clusters and those values are itself the 'Average PR time' values.
+ 
+ * These three values describe the whole dataset where the average PR active period is: 
+   * Either approx 3 days.
+   * Or approx 28 days (a month).
+   * Or approx 80 days or greater.
```

Comment:
Refactor all of this to a `EXPLANATION.md` file in the same directory. For most users they won't be concerned about getting to know how the script works.

Reply:
Yep, On it!